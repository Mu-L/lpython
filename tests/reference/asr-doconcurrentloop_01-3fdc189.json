{
    "basename": "asr-doconcurrentloop_01-3fdc189",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/doconcurrentloop_01.py",
    "infile_hash": "2fe3769863a595a01e46a88bf55c944e61a0d141d5c75816ffa74d96",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-doconcurrentloop_01-3fdc189.stdout",
    "stdout_hash": "da709ff1a274319233a59bb105289339e17c17d5ac30162a7bf269bb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}