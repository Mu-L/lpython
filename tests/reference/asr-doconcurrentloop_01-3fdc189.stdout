(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        7
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [main0]
                                    []
                                    [(SubroutineCall
                                        2 main0
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            main0:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            a:
                                                (Variable
                                                    4
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 10000 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    4
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 10000 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            c:
                                                (Variable
                                                    4
                                                    c
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 10000 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    4
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            nsize:
                                                (Variable
                                                    4
                                                    nsize
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            scalar:
                                                (Variable
                                                    4
                                                    scalar
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    main0
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [triad]
                                    []
                                    [(Assignment
                                        (Var 4 scalar)
                                        (Cast
                                            (RealConstant
                                                10.000000
                                                (Real 8)
                                            )
                                            RealToReal
                                            (Real 4)
                                            (RealConstant
                                                10.000000
                                                (Real 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 4 nsize)
                                        (ArraySize
                                            (Var 4 a)
                                            ()
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (DoConcurrentLoop
                                        []
                                        []
                                        []
                                        []
                                        [(Assignment
                                            (ArrayItem
                                                (Var 4 a)
                                                [(()
                                                (Var 4 i)
                                                ())]
                                                (Real 4)
                                                RowMajor
                                                ()
                                            )
                                            (Cast
                                                (RealConstant
                                                    5.000000
                                                    (Real 8)
                                                )
                                                RealToReal
                                                (Real 4)
                                                (RealConstant
                                                    5.000000
                                                    (Real 4)
                                                )
                                            )
                                            ()
                                        )
                                        (Assignment
                                            (ArrayItem
                                                (Var 4 b)
                                                [(()
                                                (Var 4 i)
                                                ())]
                                                (Real 4)
                                                RowMajor
                                                ()
                                            )
                                            (Cast
                                                (RealConstant
                                                    5.000000
                                                    (Real 8)
                                                )
                                                RealToReal
                                                (Real 4)
                                                (RealConstant
                                                    5.000000
                                                    (Real 4)
                                                )
                                            )
                                            ()
                                        )]
                                    )
                                    (SubroutineCall
                                        2 triad
                                        ()
                                        [((ArrayPhysicalCast
                                            (Var 4 a)
                                            FixedSizeArray
                                            DescriptorArray
                                            (Array
                                                (Real 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 10000 (Integer 4) Decimal))]
                                                DescriptorArray
                                            )
                                            ()
                                        ))
                                        ((ArrayPhysicalCast
                                            (Var 4 b)
                                            FixedSizeArray
                                            DescriptorArray
                                            (Array
                                                (Real 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 10000 (Integer 4) Decimal))]
                                                DescriptorArray
                                            )
                                            ()
                                        ))
                                        ((Var 4 scalar))
                                        ((ArrayPhysicalCast
                                            (Var 4 c)
                                            FixedSizeArray
                                            DescriptorArray
                                            (Array
                                                (Real 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 10000 (Integer 4) Decimal))]
                                                DescriptorArray
                                            )
                                            ()
                                        ))]
                                        ()
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(StringConstant
                                                "End Stream Triad"
                                                (String 1 16 () PointerString)
                                            )]
                                            FormatPythonFormat
                                            (String -1 0 () PointerString)
                                            ()
                                        )
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            triad:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            N:
                                                (Variable
                                                    3
                                                    N
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            a:
                                                (Variable
                                                    3
                                                    a
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    3
                                                    b
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            c:
                                                (Variable
                                                    3
                                                    c
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    3
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                ),
                                            scalar:
                                                (Variable
                                                    3
                                                    scalar
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    triad
                                    (FunctionType
                                        [(Array
                                            (Real 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (Array
                                            (Real 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (Real 4)
                                        (Array
                                            (Real 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 a)
                                    (Var 3 b)
                                    (Var 3 scalar)
                                    (Var 3 c)]
                                    [(Assignment
                                        (Var 3 N)
                                        (ArraySize
                                            (Var 3 a)
                                            ()
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (DoConcurrentLoop
                                        []
                                        []
                                        []
                                        []
                                        [(Assignment
                                            (ArrayItem
                                                (Var 3 c)
                                                [(()
                                                (Var 3 i)
                                                ())]
                                                (Real 4)
                                                RowMajor
                                                ()
                                            )
                                            (RealBinOp
                                                (ArrayItem
                                                    (Var 3 a)
                                                    [(()
                                                    (Var 3 i)
                                                    ())]
                                                    (Real 4)
                                                    RowMajor
                                                    ()
                                                )
                                                Add
                                                (RealBinOp
                                                    (Var 3 scalar)
                                                    Mul
                                                    (ArrayItem
                                                        (Var 3 b)
                                                        [(()
                                                        (Var 3 i)
                                                        ())]
                                                        (Real 4)
                                                        RowMajor
                                                        ()
                                                    )
                                                    (Real 4)
                                                    ()
                                                )
                                                (Real 4)
                                                ()
                                            )
                                            ()
                                        )]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        8
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    8
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        8 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                )
        })
    []
)
